src/rt/gtfsrt_update.cc

```diff
@@ -163,6 +186,10 @@ void update_run(
         // Cancel skipped stops (in_allowed = out_allowed = false).
         stp =
             stop{stop{stp}.location_idx(), false, false, false, false}.value();
+        dispatch_event_change(rtt, r, stop_idx, event_type::kArr,
```

Comment:
Overall, it seems as if `dispatch_event_changes` has to use a lot of placeholder arguments that do not give information to the reader on what is happening. We should think about using more descriptive function names, e.g. `dispatch_cancelation(rtt, r, stop_idx, event_type::kArr)` could then internally pass the correct null values. The same could be done for the other use cases of `dispatch_event_change(...)`